src/internal/data/item.go

```diff
@@ -34,24 +41,26 @@ func NewDeletedItem(itemID string) Item {
 	}
 }
 
-func NewPrefetchedItem(
+func NewUnindexedPrefetchedItem(
 	reader io.ReadCloser,
 	itemID string,
-	info details.ItemInfo,
+	modTime time.Time,
 ) Item {
-	return &prefetchedItem{
+	return &unindexedPrefetchedItem{
 		id:      itemID,
 		reader:  reader,
-		info:    info,
-		modTime: info.Modified(),
+		modTime: modTime,
 	}
 }
 
-// prefetchedItem represents a single item retrieved from the remote service.
-type prefetchedItem struct {
+// unindexedPrefetchedItem represents a single item retrieved from the remote
+// service.
+//
+// This item doesn't implement ItemInfo so it's safe to use for items like
+// metadata that shouldn't appear in backup details.
+type unindexedPrefetchedItem struct {
```

Comment:
The name here is getting a little chonky.  `Prefetched` wasn't entirely easy to grok, but worked out as we had either prefetched or non-prefetched items, so things could be extrapolated.  `UnindexedPrefetched` implies the possibility of an `IndexedPrefetched` item, and I don't know where that would get us.  Further, without any reference to an index elsewhere in this code, and no explanation in the comment, I'm not even sure what the indexing here is referring to.

I feel like this decomposition could use a re-do.  Or, at least, a renaming.  some thoughts:
```go
// basically UnindexedPrefetched
type BaseItem struct {
  id, reader, modTime
}

type ItemAndInfo struct {
  BaseItem, ItemInfo
}

type LazyItem struct {
  ItemAndInfo, Getter, Errs
}
```

Low effort go at it.  Thoughts?

Reply:
I'll come back and fix naming after I get the PR train merged